Runtime/Scripts/BaseMeshSync.Textures.cs

```diff
@@ -0,0 +1,245 @@
+                                   UnityEngine.TextureFormat.RGBA32, true))) {
+                            texture.ReadFromRenderTexture(renderTarget);
+
+                            var savePath = GetSavePath(mat, textureName);
+                            TextureData.WriteToFile(savePath, texture.Texture.EncodeToPNG());
+                        }
+
+                        // Setting the texture can fail if the asset database has not imported it yet.
+                        // If that happens, save it for later and try again:
+                        if (!SetSerializedTextureForMaterial(mat, textureName)) {
+                            pendingMaterialUpdates.Add(new Tuple<Material, string>(mat, textureName));
+
+                            // Set the texture on the material to null for now so the render texture
+                            // is not saved to file again if this gets called again before it was
+                            // set to the texture in the asset library:
+                            mat.SetTextureSafe(textureName, null);
+                        }
+                        
+                        renderTarget.Release();
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Sets the texture on the material if it exists in the AssetDatabase, otherwise it schedules a delayed call to try again when the asset might be imported.
+        /// </summary>
+        /// <param name="mat">Material to set the texture on</param>
+        /// <param name="textureName">Name of the texture in the material</param>
+        /// <returns></returns>
+        private bool SetSerializedTextureForMaterial(Material mat, string textureName) {
+            var savePath = GetSavePath(mat, textureName);
+
+            // Ensure the texture is up to date:
+            AssetDatabase.ImportAsset(savePath);
+            Texture2D savedTexture;
+
+            // LoadAssetAtPath can throw an exception if it's called during a domain backup but we have no way to detect that:
+            try {
+                savedTexture = AssetDatabase.LoadAssetAtPath<Texture2D>(savePath);
+                if (savedTexture == null) {
+                    EditorApplication.delayCall -= UpdatePendingMaterials;
+                    EditorApplication.delayCall += UpdatePendingMaterials;
+
+                    return false;
+                }
+            }
+            catch (UnityException) {
+                return false;
+            }
+
+            mat.SetTexture(textureName, savedTexture);
+
+            if (textureName == MeshSyncConstants._MetallicGlossMap ||
+                textureName == MeshSyncConstants._MaskMap) {
+                mat.EnableKeyword(MeshSyncConstants._METALLICGLOSSMAP);
+                mat.EnableKeyword(MeshSyncConstants._METALLICSPECGLOSSMAP);
+
+                TextureImporter importer = (TextureImporter)AssetImporter.GetAtPath(savePath);
```

Comment:
Relying on a filename suffix like "baked" can cause issues if users have textures with the same names.
We'd always need to import twice. `AssetImporter.GetAtPath` returns null if the asset has not been imported.
So I don't think this can really be improved, the import time is not really the issue anyway, once the texture has been sent and imported, it's never imported again.